### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR updates the following packages:

- `bash`, update from version 5.1.4 to 5.1.8;
- `base`, add `man-db` to this meta-package;
- `base-devel`, add `man-pages-posix` to this meta-package;
- `zstd`, update from version 1.4.9 to 1.5.0;
- `iana-etc`, update from version 2021-04-07 to 2021-06-11;
- `libexpat`, update from version 2.3.0 to 2.4.1;
- `m4`, update from version 1.4.18 to 1.4.19;
- `gdbm`, update from version 1.19 to 1.20;
- `curl`, update from version 7.76.0 to 7.77.0;
- `xorg-server`, update from version 1.20.11 to 1.20.12;
- `vim`, update from version 8.2.2604 to 8.2.3001;
- `nano`, update from version 5.6.1 to 5.8;
- `sqlite`, update from version 3.35.4 to 3.36.0;
- `cmake`, update from version 3.20.1 to 3.20.5;
- `harfbuzz`, update from version 2.8.0 to 2.8.2;
- `xorg-proto`, update from version 2021.3 to 2021.4;
- `libepoxy`, update from version 1.5.4 to 1.5.8;
- `libxkbcommon`, update from version 1.2.1 to 1.3.0;
- `libjpeg-turbo`, update from version 2.0.5 to 2.1.0;
- `wayland-protocols`, update from version 1.20 to 1.21;
- `mesa`, update from version 20.3.1 to 21.1.4;
- `glib`, update from version 2.68.1 to 2.68.3;
- `libtasn1`, update from version 4.16.0 to 4.17.0;
- `libXfixes`, update from version 5.0.3 to 6.0.0;
- `libxml2`, , update from version 2.9.10 to 2.9.12;
- `libX11`, update from version 1.7.0 to 1.7.2;
- `libdrm`, update from version 2.4.105 to 2.4.107;
- `p11-kit`, update from version 0.23.21 to 0.24.0;
- `libogg`, update from version 1.3.4 to 1.3.5;
- `nettle`, update from version 3.7.1 to 3.7.3;
- `xkeyboard-config`, update from version 2.32 to 2.33;
- `nss`, update from version 3.66 to 3.68;
- `ca-certificates`, update from version 20210605.3.66 to 20210714.3.68;
- `pcre`, update from version 8.44 to 8.45;
- `glu`, update from version 9.0.1 to 9.0.2;
- `libuv`, update from version 1.41.0 to 1.41.1.

Furthermore, this PR fixes the following CVE's:
- Low: [CVE-2021-22898](https://nvd.nist.gov/vuln/detail/CVE-2021-22898) in `curl`;
- Medium: [CVE-2013-0340](https://nvd.nist.gov/vuln/detail/CVE-2013-0340) in `libexpat`, [CVE-2021-22897](https://nvd.nist.gov/vuln/detail/CVE-2021-22897) in `curl`, [CVE-2021-20205](https://nvd.nist.gov/vuln/detail/CVE-2021-20205) in `libjpeg-turbo` and [CVE-2021-3541](https://nvd.nist.gov/vuln/detail/CVE-2021-3541) in `libxml2`;
- High: [CVE-2021-22901](https://nvd.nist.gov/vuln/detail/CVE-2021-22901) in `curl` and [CVE-2021-22918](https://nvd.nist.gov/vuln/detail/CVE-2021-22918) in `libuv`;
- Critical: [CVE-2021-31535](https://nvd.nist.gov/vuln/detail/CVE-2021-31535) in `libX11`, which has existed for over 35 years before fixing.

Several ports are still dragging because of various reasons, most notably on this list are: `pango`, `less`, `libinput` and `python`.
As a final note, both `harfbuzz` and `libxml2` had issues linking against `icu`, this must be investigated and support for `icu` has been temporarily disabled in `harfbuzz` and `libxml2` pending this issue.

### CODE REVIEW

File: bootstrap.d/net-misc.yml [hunk 1]
```diff
@@ -3,15 +3,15 @@ packages:
...
+      tag: 'curl-7_77_0'
+      version: '7.77.0'
       tools_required:
         - host-autoconf-v2.69
         - host-automake-v1.15
         - host-libtool
         - host-autoconf-archive
       regenerate:
-        - args: ['./buildconf']
+        - args: ['autoreconf', '-fiv']
```
Comment (avdgrinten) [inline bootstrap.d/net-misc.yml:14]: Why is this done? Did the upstream package remove that script?

File: patches/mesa/0001-Add-Managarm-support.patch [hunk 2]
```diff
@@ -47,17 +49,34 @@ index c2843a30766..61e926e9178 100644
...
+ 
++#ifndef __managarm__
+ #ifdef HAVE_DL_ITERATE_PHDR
+ #include <dlfcn.h>
+ #include <link.h>
+@@ -134,3 +135,4 @@ build_id_data(const struct build_id_note *note)
+ }
+ 
+ #endif
++#endif
```
Comment (avdgrinten) [inline patches/mesa/0001-Add-Managarm-support.patch:79]: Why is this done?

File: bootstrap.d/net-misc.yml (hunk 1 above)
Comment (Dennisbonke) [reply bootstrap.d/net-misc.yml:14]: On running `./buildconf`, a warning message is printed stating that one should use `autoreconf -fiv` instead. I assume `./buildconf` is deprecated upstream.

File: patches/mesa/0001-Add-Managarm-support.patch (hunk 2 above)
Comment (Dennisbonke) [reply patches/mesa/0001-Add-Managarm-support.patch:79]: Hmm, I don't remember exactly, and a rebuild of mesa without these lines added works too. I'll drop it from the patch.

**Review by avdgrinten — APPROVED** [review]
LGTM now.

**Pull request merged**